src/app/modules/consultations/consultation-profile/response-footer/response-footer.component.ts

```diff
@@ -0,0 +1,255 @@
+          const variables = {id: this.consultationId};
+          const resp: any = store.readQuery({query: ConsultationProfileCurrentUser, variables});
+          if (res) {
+            for (const value of resp['consultationProfile'].sharedResponses.edges) {
+              if (value.node.id ===  response['id']) {
+                if (response.votedAs) {
+                  value.node[response.votedAs.voteDirection + 'VoteCount'] -= 1;
+                }
+                value.node.votedAs = null;
+                this.response = value;
+                break;
+              }
+            }
+          }
+          store.writeQuery({query: ConsultationProfileCurrentUser, variables, data: resp});
+        }
+      })
+      .subscribe((res) => {
+        if (createVote) {
+          this.createVote(response, direction);
+        }
+      }, err => {
+        this.errorService.showErrorModal(err);
+      });
+    }
+  }
+
+  createVote(response, direction) {
+    const vote = {
+      consultationResponseVote: {
+        consultationResponseId: response.id,
+        voteDirection: direction
+      }
+    };
+    this.apollo.mutate({
+      mutation: VoteCreateQuery,
+      variables: vote,
+      update: (store, {data: res}) => {
+        const variables = {id: this.consultationId};
+        const resp: any = store.readQuery({query: ConsultationProfileCurrentUser, variables});
+        if (res) {
+          for (const value of resp['consultationProfile'].sharedResponses.edges) {
+            if (value.node.id ===  response['id']) {
+              if (value.node[res.voteCreate.voteDirection + 'VoteCount']) {
+                value.node[res.voteCreate.voteDirection + 'VoteCount'] += 1;
+              } else {
+                value.node[res.voteCreate.voteDirection + 'VoteCount'] = 1;
+              }
+              value.node.votedAs = res.voteCreate;
+              this.response = value;
+              break;
+            }
+          }
+        }
+        store.writeQuery({query: ConsultationProfileCurrentUser, variables, data: resp});
+      }
+    })
+    .subscribe((data) => {
+    }, err => {
+      this.errorService.showErrorModal(err);
+    });
+  }
+
+  toggleShareBlock(id) {
+    if (id) {
+      this.responseId = id;
+      this.showShareBlock = !this.showShareBlock;
+    }
+  }
+
+  showCreateResponse() {
+    if ((this.checkClosed(this.profileData ? this.profileData.responseDeadline : null) === 'Closed')
+        || !this.currentUser || (this.profileData && this.profileData.respondedOn)) {
+        return false;
+    }
+    return true;
+  }
+
+  checkClosed(deadline) {
+    if (deadline) {
+      const today = moment();
+      const lastDate = moment(deadline);
+      const difference = lastDate.diff(today, 'days');
+      if (difference <= 0) {
+        return difference === 0 ? 'Last day to respond' : 'Closed';
+      } else {
+        return `Active`;
+      }
+    }
+  }
+
+  questionnaireExist() {
```

Comment:
@Nishanthh I guess this is not being used anywhere?

Reply:
@avikt yes before it is not used, but used now.

Reply:
@Nishanthh I don't see it called anywhere bro